### PR TITLE
feat: migrate resource processors to transform processors and add attributes with empty values tests

### PR DESCRIPTION
**Pull request by TeodorSAP**
## Description

Changes proposed in this pull request (what was done and why):

- [x] Migrate processor: `resource/insert-cluster-attributes` -> `transform/insert-cluster-attributes`
- [x] Migrate processor: `resource/drop-kyma-attributes` -> `transform/drop-kyma-attributes`
- [x] Migrate processor: `resource/drop-service-name` -> `transform/drop-service-name`
- [x] Migrate processor: `resource/drop-skip-enrichment-attribute` -> `transform/drop-skip-enrichment-attribute`
- [x] Migrate processor: `resource/set-kyma-input-name-istio` -> `transform/set-kyma-input-name-istio`
- [x] Migrate processor: `resource/set-kyma-input-name-prometheus` -> `transform/set-kyma-input-name-prometheus`
- [x] Migrate processor: `resource/set-kyma-input-name-runtime` -> `transform/set-kyma-input-name-runtime`
- [x] Migrate processor: `resource/set-kyma-input-name-kyma` -> `transform/set-kyma-input-name-kyma`
- [x] Migrate processor: `resource/set-kyma-input-name-otlp` -> `transform/set-kyma-input-name-otlp`
- [x] Implement UT & E2E tests for covering empty value attributes edge-case
- [x] Remove resource processor definitions and mentions

Changes refer to particular issues, PRs or documents:

- https://github.com/kyma-project/telemetry-manager/issues/1412

## Traceability
- [ ] The PR is linked to a GitHub issue.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [ ] If the change is user-facing, the documentation has been adjusted.
- [ ] If a CRD is changed, the corresponding Busola ConfigMap has been adjusted.
- [ ] The feature is unit-tested.
- [ ] The feature is e2e-tested.

### CODE REVIEW

File: internal/otelcollector/config/common/types.go [hunk 1]
```diff
@@ -178,10 +178,6 @@ type PodAssociation struct {
 	Name string `yaml:"name,omitempty"`
 }
 
-type ResourceProcessor struct {
-	Attributes []AttributeAction `yaml:"attributes"`
-}
-
 type AttributeAction struct {
```
[the remark this answers is not in the document]
Comment (TeodorSAP) [reply internal/otelcollector/config/common/types.go]: No, removed. Thanks for noticing.

File: internal/otelcollector/config/common/processor_builders.go [hunk 2]
```diff
@@ -285,6 +233,32 @@ func TransformSpecsToProcessorStatements(specs []telemetryv1alpha1.TransformSpec
 	return result
 }
 
+// InsertClusterAttributesProcessorStatements creates processor statements for the transform processor that inserts cluster attributes
+func InsertClusterAttributesProcessorStatements(clusterName, clusterUID, cloudProvider string) []TransformProcessorStatements {
```
[the remark this answers is not in the document]
Comment (TeodorSAP) [reply internal/otelcollector/config/common/processor_builders.go]: Lmk if you are fine with where I placed the ClusterOptions struct. Or if you have a better idea.
